libs/remote_storage/src/azure_blob.rs

```diff
@@ -323,10 +326,41 @@ impl RemoteStorage for AzureBlobStorage {
         Ok(())
     }
 
-    async fn copy(&self, _from: &RemotePath, _to: &RemotePath) -> anyhow::Result<()> {
-        Err(anyhow::anyhow!(
-            "copy for azure blob storage is not implemented"
-        ))
+    async fn copy(&self, from: &RemotePath, to: &RemotePath) -> anyhow::Result<()> {
+        let _permit = self.permit(RequestKind::Copy).await;
+        let blob_client = self.client.blob_client(self.relative_path_to_name(to));
+
+        let source_url = format!(
+            "{}/{}",
+            self.client.url()?,
+            self.relative_path_to_name(from)
+        );
+        let builder = blob_client.copy(Url::from_str(&source_url)?);
+
+        let result = builder.into_future().await?;
+
+        let mut copy_status = result.copy_status;
+        loop {
```

Comment:
nit: Perhaps we could give up and return an error after a certain amount of time (say 60 seconds). If a copy takes longer than that, it's likely something is wrong.

Reply:
Yeah our objects are a few hundred MB in size at most, which means that it would be strange if copying them takes a non trivial amount of time. I will change it to have a max duration and then bail out if it is reached.